### PR TITLE
Hotfix/filter eager loading fetch eager

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | na
| License       | MIT
| Doc PR        | na

ping @teohhanhui

### CODE REVIEW

File: tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php [hunk 1]
```diff
@@ -270,6 +347,6 @@ public function testCompositeIdentifiers()
 
     private function toDQLString(string $dql): string
     {
-        return preg_replace('/\\r\\n|\\n/', '', str_replace('  ', '', $dql));
+        return preg_replace(['/\s+/', '/\(\s/', '/\s\)/'], [' ', '(', ')'], $dql);
```
Comment (soyuka) [inline tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php]: This should be better \o/

Comment (teohhanhui) [inline tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php]: Actually this is unsafe... It might change whitespace within a SQL literal (string). Lol...

File: src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php [hunk 2]
```diff
@@ -143,4 +146,27 @@ private function isForceEager(string $resourceClass, array $options): bool
 
         return is_bool($forceEager) ? $forceEager : $this->forceEager;
     }
+
+    private function hasFetchEagerAssociation(EntityManager $em, ClassMetadataInfo $classMetadata, &$checked = [])
```
Comment (teohhanhui) [inline src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php]: Is it possible to extract this logic out for reuse between `EagerLoadingExtension` and `FilterEagerLoadingExtension`?

File: tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php (hunk 1 above)
Comment (soyuka) [inline tests/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtensionTest.php]: Of course, as said before it makes tests more readable :).

File: src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php (hunk 2 above)
Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php]: I don't want to to this to avoid a 2-pass recursive call in `EagerLoadingExtension`. If we want to extract this logic we need to add cache. I obviously agree that we should add a new class with the `shouldEagerLoad` logic.

Comment (dunglas) [inline src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php]: Can't this be extracted in another PR?

Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php]: Yeah it can and should, +1 to do this in another PR as it's not a priority.

**Comment by soyuka** [comment]
Feel free to merge I have a note that I should refactor the `shouldForceEager` thing.

**Comment by dunglas** [comment]
Thank you @soyuka
